src/testUtils/mocks/dataSources.ts

```diff
@@ -1468,20 +1468,22 @@ function isOption<T extends Codec>(codec: any): codec is Option<T> {
   return typeof codec?.unwrap === 'function';
 }
 
+type MockCodec<C extends Codec> = C & { eq: sinon.SinonStub };
+
 /**
  * @hidden
  */
-const createMockCodec = (codec: unknown, isEmpty: boolean): Codec => {
-  const clone = cloneDeep(codec) as Mutable<Codec>;
+const createMockCodec = (codec: unknown, isEmpty: boolean): MockCodec<Codec> => {
```

Comment:
You forgot to actually add the`eq` stub. Also, remove it from `createMockStringCodec` because it would be redundant (I suspect your tests passed because it was there)